sim/mcuboot-sys/src/api.rs

```diff
@@ -1,55 +1,94 @@
 //! HAL api for MyNewt applications
 
-use simflash::{Result, Flash};
+use simflash::{Result, Flash, FlashPtr};
 use libc;
 use log::LogLevel;
 use std::mem;
 use std::slice;
+use std::collections::HashMap;
+use std::sync::Mutex;
+use std::ops::Deref;
 
-// The current active flash device.  The 'static is a lie, and we manage the lifetime ourselves.
-static mut FLASH: Option<*mut Flash> = None;
+/// A FlashMap maintain a table of [device_id -> Flash trait]
+pub type FlashMap = HashMap<u8, FlashPtr>;
+
+lazy_static! {
+    static ref FLASH: Mutex<FlashMap> = {
+        Mutex::new(HashMap::new())
+    };
+}
+
+lazy_static! {
+    /// Maintain a table of [device_id -> (align, erased_val)]
+    static ref FLASH_PARAMS: Mutex<HashMap<u8, (u8, u8)>> = {
```

Comment:
I'm kind of on the fence as to whether this map should map to a struct with two fields rather than a general tuple. It does seem to currently only be used below, so isn't that great of an issue. The compiler will generate the same code with a struct defined.

Reply:
Oh yeah, it definitely improves a lot the legibility; I squashed the fix into the original commit. Thanks!